clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/AwsCleanupProvider.groovy

```diff
@@ -19,23 +19,15 @@ package com.netflix.spinnaker.clouddriver.aws.provider
 import com.netflix.spinnaker.cats.agent.Agent
 import com.netflix.spinnaker.cats.agent.AgentSchedulerAware
 import com.netflix.spinnaker.cats.provider.Provider
+import com.netflix.spinnaker.clouddriver.security.BaseProvider
+import org.springframework.stereotype.Component
 
-class AwsCleanupProvider extends AgentSchedulerAware implements Provider {
+@Component
+class AwsCleanupProvider extends BaseProvider implements Provider {
```

Comment:
`BaseProvider` already implements `Provider`.

Reply:
Also it probably shouldn't be created as a `Component` but in the configuration class as a Bean (as currently) since it is controlled by `aws.enabled`